dll/win32/msctf/mlng.cpp

```diff
@@ -0,0 +1,585 @@
+/*
+ * PROJECT:     ReactOS msctf.dll
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Multi-language handling of Cicero
+ * COPYRIGHT:   Copyright 2024 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include <stdlib.h>
+
+#define WIN32_NO_STATUS
+#define COBJMACROS
+#define INITGUID
+#define _EXTYPES_H
+
+#include <windows.h>
+#include <imm.h>
+#include <imm32_undoc.h>
+#include <cguid.h>
+#include <tchar.h>
+#include <msctf.h>
+#include <ctffunc.h>
+#include <shlwapi.h>
+#include <strsafe.h>
+
+#include <cicreg.h>
+#include <cicarray.h>
+
+#include <wine/debug.h>
+
+#include "mlng.h"
+
+WINE_DEFAULT_DEBUG_CHANNEL(msctf);
+
+extern CRITICAL_SECTION g_cs;
+
+CicArray<MLNGINFO> *g_pMlngInfo = NULL;
+INT CStaticIconList::s_cx = 0;
+INT CStaticIconList::s_cy = 0;
+CStaticIconList g_IconList;
+
+/***********************************************************************
+ * The helper funtions
+ */
+
+static VOID GetLocaleInfoString(HKL hKL, LPWSTR pszDesc, UINT cchDesc)
+{
+    if (!::GetLocaleInfoW(LOWORD(hKL), LOCALE_SLANGUAGE, pszDesc, cchDesc))
+        *pszDesc = UNICODE_NULL;
+#if 0
+    if (HIWORD(hKL) != LOWORD(hKL))
+        GetKbdLayoutName(hKL, pszDesc, cchDesc);
+#endif
+}
+
+/// @unimplemented
+static HKL GetSubstitute(HKL hKL)
+{
+    return hKL;
+}
+
+/// @implemented
```

Comment:
Why 3 slashes instead of two? Same question for line 55.

Reply:
It's for Doxygen documentation.